src/Bridge/Doctrine/Persister/ObjectManagerPersister.php

```diff
@@ -74,4 +99,14 @@ private function getPersistableClasses(ObjectManager $manager): array
 
         return $persistableClasses;
     }
+
+    private function getMetadata(string $class): ClassMetadataInfo
```

Comment:
It's failing for MongoDB, wrong type.

We can return `Doctrine\Common\Persistence\Mapping\ClassMetadata` here (which is correct), but that doesn't guarantee us `$metadata->idGenerator` which we use to fetch the current generator (to be able to reset it afterward).

Reply:
Hm how does it works for MongoDB then? Isn't there the concept of generator there still with the same issue?

Reply:
It should work, but we can't link against the interface because it doesn't provide access to the current generator.